### PR TITLE
Support for defining SC at container level

### PR DESCRIPTION
**Pull request by AsfaMumtaz**
Added support for defining Security Context at Container level for deployments.
**Example**:
deployment:
    containerSecurityContext:
          runAsNonRoot: false

**Note: By default runAsNonRoot and readOnlyRootSystem is true.**

### CODE REVIEW

**Comment by stakater-user** [comment]
@AsfaMumtaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/application/commit/00397842f80764a73d38e50c79517e6e253e87ef/checks) has Failed!

**Comment by stakater-user** [comment]
@AsfaMumtaz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/application/commit/ad742934fdd8a0e1dce4b4a479b51011d0964d38/checks) has Failed!

File: application/values.yaml [hunk 1]
```diff
@@ -224,9 +225,16 @@ deployment:
   # - name: sidecar-contaner
   #   image: busybox
   #   imagePullPolicy: IfNotPresent
-  #   command: ['/bin/sh']    
+  #   command: ['/bin/sh']
+
+  # Security Context at Container Level
+  #  containers:
+  #    securityContext:
```
Comment (rasheedamir) [inline application/values.yaml:232]: @AsfaMumtaz lets add enable flag and we should have enabled by default

**Comment by rasheedamir** [comment]
@AsfaMumtaz update changelog as well

**Comment by stakater-user** [comment]
@AsfaMumtaz Validation Successful

File: application/templates/deployment.yaml [hunk 2]
```diff
@@ -233,6 +233,14 @@ spec:
           requests:
             memory: {{ .Values.deployment.resources.requests.memory }}
             cpu: {{ .Values.deployment.resources.requests.cpu }}
+        {{- if .Values.deployment.containerSecurityContext }}
+        securityContext:
+{{ toYaml .Values.deployment.containerSecurityContext | indent 10 }}
+        {{- else }}
+        securityContext:
+          readOnlyRootFilesystem: true
+          runAsNonRoot: true
```
Comment (rasheedamir) [inline application/templates/deployment.yaml:242]: @AsfaMumtaz why hardcode here just two values?

**Review by rasheedamir — APPROVED** [review]

**Pull request merged**